src/core/ui/components/drop-down/index.tsx

```diff
@@ -0,0 +1,353 @@
+import { useEffect, useMemo, useRef, useState } from 'react';
+import { ScaleLoader } from 'react-spinners';
+import { useCoreSlashAuth } from '../../context/core-slashauth';
+import { shortenEthAddress } from '../../../../shared/utils/eth';
+import { profilePicturePlaceholder } from './profilePicturePlaceholder';
+import { chevronDown } from './chevronDown';
+import { googleIcon } from './googleIcon';
+import { twitterIcon } from './twitterIcon';
+import { copyIcon } from './copyIcon';
+import { plusIcon } from './plusIcon';
+import { gearIcon } from './gearIcon';
+import { logoutIcon } from './logoutIcon';
+
+type TestUser = {
+  name?: string;
+  loggedIn: boolean;
+  wallet?: { default: string };
+  email?: string;
+  social?: {
+    google: string;
+    twitter: string;
+  };
+};
+export const DropDown = () => {
+  const user: TestUser = testUser.walletOnly;
+  const [isOpen, setIsOpen] = useState(false);
+  const context = useCoreSlashAuth();
+  const { isReady, logout, openSignIn, connectWallet } = context;
+  const [wallet, walletDisplay] = useMemo(() => {
+    if (user.wallet) {
+      const address: string = user.wallet.default.split(':')[1]; //undefined possible
+      const walletDisplay = shortenEthAddress(address);
+      return [address, walletDisplay];
+    }
+    return [];
+  }, [user.wallet]);
+  const ref = useRef();
+  useEffect(() => {
```

Comment:
Can you add a comment here? I don't understand what this is doing.

Is this to close the div if it isn't clicked on?

Reply:
updated

Reply:
I'm curious why to do it this way instead of just comparing the click coords to the container rectangle from `ref`? This is a very cool idea (although, it could mean walking an entire dom tree which is a lot of steps potentially? Not sure how deep dom trees go)